src/zepben/evolve/streaming/grpc/connect.py

```diff
@@ -127,8 +166,17 @@ def _conn(host: str = "localhost", rpc_port: int = 50051, conf_address: str = No
             ) if channel_credentials else call_credentials
             channel = grpc.aio.secure_channel(f"{host}:{rpc_port}", composite_credentials)
         elif client_id and username and password:
+
+            # parsing the conf_address to host port and path
+            parsed_url = urlparse(
+                conf_address
+            )
+            host = parsed_url.hostname
+            port = parsed_url.port
+            path = parsed_url.path
+
             # Create a basic ClientCredentials authenticator
-            token_fetcher = create_token_fetcher(conf_address, verify_auth_certificates, conf_ca_filename=conf_ca, auth_ca_filename=auth_ca)
+            token_fetcher = create_token_fetcher(host, port, path, verify_auth_certificates, conf_ca_filename=conf_ca, auth_ca_filename=auth_ca)
```

Comment:
You'll also need to do this down below on line 199. Note you may want to move your parsed_url code up before this if block so you can use it in both places.

Reply:
Right, this parsed_url change also need to be applied for the case using client_secret.